### PR TITLE
Always send a 204 if the stream is empty

### PR DESCRIPTION
**Pull request by dmathieu**

### CODE REVIEW

**Comment by dmathieu** [comment]
@tevanoff @heroku/build I've figured it out. `NoContent` was only checking on redis, but when the stream is archived, it's not there anymore and we'd always ignore the method.

I've updated the behavior to look for how many bytes were sent in `io.Copy`, and send a 204 if that was 0.

**Comment by dmathieu** [comment]
[io.Copy](https://golang.org/pkg/io/#Copy)'s first argument is the number of bytes written. So that assumption seems safe to assume 😜

File: server/endpoints.go [hunk 1]
```diff
@@ -105,6 +105,11 @@ func (s *Server) subscribe(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		logError(r, err)
 	}
+	if n == 0 {
+		handleError(w, r, errNoContent)
```
Comment (tt) [inline server/endpoints.go]: Treating this as an error is a bit odd.

Comment (dmathieu) [reply server/endpoints.go]: Indeed. I've fixed that in #165.

**Pull request merged**